infer.py

```diff
@@ -88,33 +89,52 @@ def infer(self, model, output_path, args):
                     sliced_data = data
                 self._visualize_attention(model, args.beam_size, args.output_history, sliced_data, args.res1, args.res2, args.res3, output)
 
-    def _inner_infer(self, model, beam_size, output_history, sliced_orig_data, sliced_preproc_data, output):
-        for i, (orig_item, preproc_item) in enumerate(
-                tqdm.tqdm(zip(sliced_orig_data, sliced_preproc_data),
-                          total=len(sliced_orig_data))):
-            beams = beam_search.beam_search(
-                    model, orig_item, preproc_item, beam_size=beam_size, max_steps=1000)
+    def _inner_infer(self, model, beam_size, output_history, sliced_orig_data, sliced_preproc_data, output, nproc):
+        list_items = list(enumerate(zip(sliced_orig_data, sliced_preproc_data)))
+        total = len(sliced_orig_data)
 
-            decoded = []
-            for beam in beams:
-                model_output, inferred_code = beam.inference_state.finalize()
+        params = []
+        pbar_queue = Manager().Queue()
```

Comment:
```suggestion
        pbar_queue = multiprocessing.Manager().Queue()
```

Reply:
Can we change this so that `multiprocessing` is used only when `nproc > 1`? Always using multiprocessing makes it harder to use debuggers with the code.

Reply:
fixed